engine/src/test/java/io/camunda/zeebe/engine/state/instance/EventScopeInstanceStateTest.java

```diff
@@ -7,303 +7,542 @@
-import org.junit.Before;
-import org.junit.Rule;
-import org.junit.Test;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
 
+@ExtendWith(ZeebeStateExtension.class)
 public final class EventScopeInstanceStateTest {
 
-  @Rule public final ZeebeStateRule stateRule = new ZeebeStateRule();
+  private static final int SCOPE_KEY = 123;
+  private static final Collection<DirectBuffer> NO_INTERRUPTING_ELEMENT_IDS =
+      Collections.emptySet();
+  private static final Collection<DirectBuffer> NO_BOUNDARY_ELEMENT_IDS = Collections.emptySet();
+
+  private MutableZeebeState zeebeState;
 
   private MutableEventScopeInstanceState state;
 
-  @Before
-  public void setUp() {
-    final MutableZeebeState zeebeState = stateRule.getZeebeState();
+  @BeforeEach
+  void setUp() {
     state = zeebeState.getEventScopeInstanceState();
   }
 
   @Test
-  public void shouldCreateInterruptingEventScopeInstance() {
+  void shouldCreateInterruptingEventScopeInstance() {
+    // given
+    final var interruptingElementId = wrapString("interrupting");
+    final var interruptingElementIds = Collections.singleton(interruptingElementId);
+
+    // when
+    state.createInstance(SCOPE_KEY, interruptingElementIds, NO_BOUNDARY_ELEMENT_IDS);
+
+    // then
+    final EventScopeInstance instance = state.getInstance(SCOPE_KEY);
+    assertThat(instance.isAccepting()).isTrue();
+    assertThat(instance.isInterruptingElementId(interruptingElementId)).isTrue();
+    assertThat(instance.isBoundaryElementId(interruptingElementId)).isFalse();
+  }
+
+  @Test
+  void shouldCreateNonInterruptingEventScopeInstance() {
+    // given
+    final var nonInterruptingElementId = wrapString("non-interrupting");
+
+    // when
+    state.createInstance(SCOPE_KEY, NO_INTERRUPTING_ELEMENT_IDS, NO_BOUNDARY_ELEMENT_IDS);
+
+    // then
+    final EventScopeInstance instance = state.getInstance(SCOPE_KEY);
+    assertThat(instance.isAccepting()).isTrue();
+    assertThat(instance.isInterruptingElementId(nonInterruptingElementId)).isFalse();
+    assertThat(instance.isBoundaryElementId(nonInterruptingElementId)).isFalse();
+  }
+
+  @Test
+  void shouldCreateEventScopeInstanceWithBoundaryEvents() {
+    // given
+    final var nonInterruptingBoundaryElementId = wrapString("non-interrupting-boundary");
+    final var interruptingBoundaryElementId = wrapString("interrupting-boundary");
+    final var boundaryElementIds =
+        Set.of(nonInterruptingBoundaryElementId, interruptingBoundaryElementId);
+    final var interruptingElementIds = Set.of(interruptingBoundaryElementId);
+
+    // when
+    state.createInstance(SCOPE_KEY, interruptingElementIds, boundaryElementIds);
+
+    // then
+    final EventScopeInstance instance = state.getInstance(SCOPE_KEY);
+    assertThat(instance.isAccepting()).isTrue();
+
+    assertThat(instance.isInterruptingElementId(nonInterruptingBoundaryElementId)).isFalse();
```

Comment:
:wrench: Might make sense to extend our assertions for `EventScopeInstance` to write better assertions.

Reply:
Maybe. But I don't see it in the scope of this PR.